### DISCUSSION TITLE
MuseScore unexpected shutdown

### DISCUSSION THREAD

**Discussion by Mikjazz** [musescore/MuseScore#20631] [Q&A]
Hello everyone!

Unfortunately I am encountering this problem even more frequently when closing a project and especially now that I have made the latest update. Six times out of ten, when I save and close MuseScore, the annoying window appears informing about the unexpected closure of the program. I use MuseScore on my MacBook and don't remember having this problem with previous versions. Has this happened to anyone and if so, is there a solution?
I always thank you

**Comment by cbjeukendrup** [comment]
> the annoying window appears informing about the unexpected closure of the program

Could you please copy the content of that window to a text file, and upload that here? The information in that dialog is namely crucial in understanding why it crashes.

**Comment by MarcSabatella** [comment]
Also please ZIP and attach the score you are having trouble with, and give precise steps (as precise as possible, anyhow) as to how to reproduce the problem.

**Comment by Mikjazz** [comment]
Sure, I can always reproduce the crash with these steps:

1. I open the Test.mscz file by clicking on it.
2. I send the score to execution (Play).
3. While it's still running, I close MuseScore with command+Q
4. Crash!

Furthermore, the crash also occurs in other different ways and I can't reproduce the steps: sometimes it crashes and sometimes it doesn't.

Thank you for your help.

Marco


[Test.mscz.zip](https://github.com/musescore/MuseScore/files/13753150/Test.mscz.zip)
<img width="985" alt="Window" src="https://github.com/musescore/MuseScore/assets/123240280/eb4f5b2c-f37e-4518-b449-c596d3ab4b1b">
[Content_window.odt](https://github.com/musescore/MuseScore/files/13753163/Content_window.odt)

**Comment by cbjeukendrup** [comment]
It looks like https://github.com/musescore/MuseScore/pull/14790 might potentially fix this. In 30 minutes, a test build from this fix will be ready. You can find instructions here about how to download an run this build: https://github.com/musescore/MuseScore/wiki/Downloading-and-running-test-builds#downloading-builds-from-pull-requests
It would be great if you could try it out and let me know if that solves it indeed!

**Comment by Mikjazz** [comment]
I downloaded and followed this build. I followed the same steps to create the crash. 
Unfortunately the crash occurred promptly and I didn't solve the problem. :((

1. I open the Test.mscz file by clicking on it.
2. I send the score to execution (Play).
3. While it's still running, I close MuseScore with command+Q
4. Crash!

Thanks anyway for your availability and help

**Reply by cbjeukendrup** [reply]
Ah, too bad. Anyway, that build does fix another crash that sometimes occurred while quitting, so perhaps once that one is solved, we'll start seeing the one you are experiencing in debug builds. Eventually someone will find the cause!

**Comment by Mikjazz** [comment]
The cause has been discovered!
After several attempts I discovered the exact cause of the crash when MuseScore exited.
To avoid the crash, **before closing MuseScore**, I have to **press the space bar**. This avoids crashing!
I believe that the block is due to the fact that the "Play" function remains active and if this is not closed with the Stop command (space bar), the program blocks.
I don't know if there is a way to correct this malfunction.

**Comment by Mikjazz** [comment]
Still no news regarding this bug? Unfortunately I didn't solve it even with the new update...

**Reply by cbjeukendrup** [reply]
No news yet unfortunately. When we fix it, we'll make sure to notify you!

**Reply by Mikjazz** [reply]
Thank you so much! I'm waiting...
